src/minimizer_index_helper.hpp

```diff
@@ -0,0 +1,254 @@
+/*
+minimizer_index_helper.hpp
+Shared logic for minimizer index construction.
+*/
+#pragma once
+#include <fstream>
+#include <functional>
+#include <gbwtgraph/gbz.h>
+#include <gbwtgraph/index.h>
+#include <omp.h>
+#include <string>
+#include <type_traits>
+#include <vector>
+
+#include "snarl_distance_index.hpp"
+#include "vg/io/vpkg.hpp"
+#include "zip_code.hpp"
+
+namespace mi_helper {
+
+using namespace std;
+using namespace vg;
+
+inline size_t trailing_zeros(size_t value) {
+  size_t result = 0;
+  if (value == 0) {
+    return result;
+  }
+  while ((value & 1) == 0) {
+    value >>= 1;
+    result++;
+  }
+  return result;
+}
+
+inline size_t estimate_hash_table_size(const gbwtgraph::GBZ &gbz,
```

Comment:
If this isn't a function that's short and called often, it probably shouldn't be `inline`; it should be declared in the header and defined in a cpp file. While it's *possible* to get out of making a cpp file for a unit of code by just making all the functions in it either templates or `inline`, it's not advisable. It means the compiler has to go through the implementation code every time someone wants to have access to the interface, when it wouldn't otherwise need to.